### ISSUE TITLE
`npm i` fails with node#6.2.1 and npm#3.9.3

### ISSUE THREAD

**Issue by dopeddude** [joyent/node-bunyan-syslog#39]
Output for `$ npm i bunyan-syslog`

```
> bunyan-syslog@0.3.1 install /home/app/some-directory
> node-gyp rebuild

make: Entering directory '/home/app/some-directory/node_modules/bunyan-syslog/build'
  CXX(target) Release/obj.target/syslog/src/syslog.o
In file included from ../src/syslog.cc:7:0:
../node_modules/nan/nan.h:602:20: error: variable or field 'AddGCEpilogueCallback' declared void
       v8::Isolate::GCEpilogueCallback callback
                    ^
../node_modules/nan/nan.h:602:7: error: 'GCEpilogueCallback' is not a member of 'v8::Isolate'
       v8::Isolate::GCEpilogueCallback callback
       ^
../node_modules/nan/nan.h:603:18: error: expected primary-expression before 'gc_type_filter'
     , v8::GCType gc_type_filter = v8::kGCTypeAll) {
                  ^
../node_modules/nan/nan.h:608:20: error: variable or field 'RemoveGCEpilogueCallback' declared void
       v8::Isolate::GCEpilogueCallback callback) {
                    ^
../node_modules/nan/nan.h:608:7: error: 'GCEpilogueCallback' is not a member of 'v8::Isolate'
       v8::Isolate::GCEpilogueCallback callback) {
       ^
../node_modules/nan/nan.h:613:20: error: variable or field 'AddGCPrologueCallback' declared void
       v8::Isolate::GCPrologueCallback callback
                    ^
../node_modules/nan/nan.h:613:7: error: 'GCPrologueCallback' is not a member of 'v8::Isolate'
       v8::Isolate::GCPrologueCallback callback
       ^
../node_modules/nan/nan.h:614:18: error: expected primary-expression before 'gc_type_filter'
     , v8::GCType gc_type_filter = v8::kGCTypeAll) {
                  ^
../node_modules/nan/nan.h:619:20: error: variable or field 'RemoveGCPrologueCallback' declared void
       v8::Isolate::GCPrologueCallback callback) {
                    ^
../node_modules/nan/nan.h:619:7: error: 'GCPrologueCallback' is not a member of 'v8::Isolate'
       v8::Isolate::GCPrologueCallback callback) {
       ^
make: *** [syslog.target.mk:88: Release/obj.target/syslog/src/syslog.o] Error 1
make: Leaving directory '/home/app/some-directory/node_modules/bunyan-syslog/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
gyp ERR! System Linux 4.4.8-boot2docker
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/app/some-directory/node_modules/bunyan-syslog
gyp ERR! node -v v6.2.1
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.12
npm ERR! Linux 4.4.8-boot2docker
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "i" "bunyan-syslog"
npm ERR! node v6.2.1
npm ERR! npm  v3.9.3
npm ERR! code ELIFECYCLE

npm ERR! bunyan-syslog@0.3.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bunyan-syslog@0.3.1 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the bunyan-syslog package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs bunyan-syslog
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls bunyan-syslog
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/app/some-directory/npm-debug.log
```

with `$ npm version`

```
{ 
  npm: '3.9.3',
  ares: '1.10.1-DEV',
  http_parser: '2.7.0',
  icu: '57.1',
  modules: '48',
  node: '6.2.1',
  openssl: '1.0.2h',
  uv: '1.9.1',
  v8: '5.0.71.52',
  zlib: '1.2.8' 
}
```

**Comment by tommiv** [comment]
Sorry, but what is the status of this? Just tried to install `npm install --save bunyan-syslog` and got the same node-gyp error on 

```
⇒  node --version
v6.5.0

⇒  npm --version
3.10.3
```

I see a merged MR, but ticket is still open.

**Comment by dopeddude** [comment]
@tommiv The relevant PR "https://github.com/mcavage/node-bunyan-syslog/pull/41" is merged into the #master branch. But, the latest code is yet to be published on npm with a version bump.

@mcavage can you please bump the version and publish the latest on the npm.

@tommiv For the quick fix you can update the package.json of your project to refer to the "github#master branch or the specific commit id on the master branch of the node-bunyan-syslog repo".

**Comment by cuteboi** [comment]
This was hurting my development time for the last 4 hours. Just found this as the solution, I am using

```
npm install mcavage/node-bunyan-syslog
```

for the meantime

**Comment by ChopperLee2011** [comment]
have the same issue with Node v6, mainly because the `nan` version in used is not updated to support the latest Nodejs. 
Try to use the master branch could work, so when will the new release come out ?

**Comment by riggerthegeek** [comment]
Is there any estimate of when this fix will be pushed to npm?

**Comment by sebastianhaas** [comment]
Would also love to have a new release. 
Meanwhile, 
```shell
npm install --save github:mcavage/node-bunyan-syslog#9bdb9eb62a763c7d30e032a619138032d5ff8771
```
seems to work for me.

**Comment by melloc** [comment]
I've published `bunyan-syslog@0.3.2`, which has a newer `nan` and has been tested with node v6 and v8.
